Lib/typing.py

```diff
@@ -2763,7 +2763,12 @@ def _make_nmtuple(name, types, module, defaults = ()):
 class NamedTupleMeta(type):
 
     def __new__(cls, typename, bases, ns):
-        assert bases[0] is _NamedTuple
+        assert _NamedTuple in bases
+        for base in bases:
+            if base is not _NamedTuple and base is not Generic:
+                raise TypeError('can only inherit from a NamedTuple type '
+                                'and Generic')
```

Comment:
I would honestly just violate the line length limit and keep the entire error message string on one line.